src/components/map/actions/measure-action.js

```diff
@@ -1,77 +1,160 @@
+    });
+
+    this.removeFeatureHandler = this.layer.getSource().on('removefeature', (event) => {
+      this._handleRemoveFeature(event);
+    });
+
+    super.activate(this);
+  }
+
+  _setMeasurementTooltipsClosable(closable) {
+    this.measurementTooltips.forEach((tooltip) => {
+      // Check if tooltip still exists
+      if (tooltip) {
+        if (tooltip.getElement()) {
+          if (closable) {
+            tooltip.getElement().setAttribute('data-vl-closable', closable);
+          } else {
+            tooltip.getElement().removeAttribute('data-vl-closable');
+          }
+        }
+      }
+    });
+  }
+
+  _showMeasurementTooltip(feature, tooltipOverlay, tooltipElement) {
+    const length = feature.getGeometry().getLength().toFixed(2);
+    tooltipElement.textContent = `${length} m`;
+    tooltipOverlay.setElement(tooltipElement);
+    tooltipOverlay.setPosition(feature.getGeometry().getLastCoordinate());
+  }
+
+  _handleDrawStart(event) {
+    // Add measurement line (feature) and tooltip (overlay)
+
+    const id = this.featureCounter;
+    this.featureCounter += 1;
+
+    this._setMeasurementTooltipsClosable(false); // Hide close buttons on tooltips while drawing
+
+    const { feature } = event;
+    feature.setId(id);
+
+    const tooltipElement = document.createElement('vl-pill');
+    tooltipElement.setAttribute('data-vl-opacity', '0.8');
+
+    tooltipElement.addEventListener(
+      'close',
+      (closePillEvent) => {
+        this._handleRemoveMeasurement(closePillEvent, feature);
+      },
+      { once: true },
+    );
+
+    const tooltipOverlay = new Overlay({
+      offset: [0, 40],
+      positioning: 'bottom-center',
+      stopEvent: true,
+      insertFirst: true,
     });
 
-    const removeTooltip = (id) => {
-      this.map.removeOverlay(this.measureTooltips[id]);
-      this.measureTooltips[id] = null;
-    };
+    this.map.addOverlay(tooltipOverlay);
+
+    this.measurementTooltips[id] = tooltipOverlay;
 
-    this.layer.getSource().on('removefeature', (event) => {
-      removeTooltip(event.feature.getId());
+    this.measurePointermoveHandler = this.map.on('pointermove', () => {
+      this._showMeasurementTooltip(feature, tooltipOverlay, tooltipElement);
     });
 
-    this.cleanUp = () => {
-      unByKey(this.measurePointermoveHandler);
+    this.drawInteraction.on(
+      'drawend',
+      () => {
+        this._setMeasurementTooltipsClosable(true);
+      },
+      { once: true },
+    );
+  }
+
+  _removeMeasureFeature(feature) {
+    const source = this.layer.getSource();
+    if (feature && (feature.getId() == null || source.getFeatureById(feature.getId()) === feature)) {
+      source.removeFeature(feature);
+      this.map.render();
+    }
+  }
+
+  _removeMeasurementTooltip(id) {
+    this.map.removeOverlay(this.measurementTooltips[id]);
+    this.measurementTooltips[id] = null; // Not removing the item from the list because id corresponds with list index
```

Comment:
Ik zou index niet verwarren met ids. Je hebt een array `measurementTooltips`, dit is gewoon een array met tooltips met ids, als je er een wil toevoegen spread je er een tooltip bij, als je er een wil verwijderen kan je je array filteren.

Reply:
Dit systeem was er al gewoon hoor, ik heb het gewoon verder gebruikt. Maar als het echt stoort zal ik 't anders doen.